networking_bigswitch/bsnlldp/send_lldp.py

```diff
@@ -294,6 +294,106 @@ def daemonize():
         os._exit(os.EX_OK)
 
 
+PCI_IDS_DIR = "/run/pci_ids"
+# read lldp status for different interfaces
+lldp_status = {}
+
+def diff_in_lists(first_list, second_list):
+    second = set(second_list)
+    return [item for item in first_list if item not in second]
+
+def find_pci_id(uplink):
+    if os.path.exists("/sys/bus/pci/devices/%s" % uplink):
+        return uplink
+
+    stash = PCI_IDS_DIR + "/" + uplink
+    if os.path.exists(stash):
+        return readfile(stash)
+
+    if not os.path.exists("/sys/class/net/%s" % uplink):
+        raise RuntimeError("No such network device %s" % uplink)
+
+    pci_id = os.path.basename(os.readlink("/sys/class/net/%s/device" % uplink))
+
+    if not os.path.exists(PCI_IDS_DIR):
+        os.mkdir(PCI_IDS_DIR)
+
+    writefile(stash, pci_id)
+    return pci_id
+
+
+# read the lldp tx status for interfaces
+def read_lldp_status_from_device(intfs):
+    lldp_start_str = "lldp start"
+    lldp_stop_str = "lldp stop"
+
+    LOG.syslog("Read LLDP Tx status from device as needed")
+    for intf in intfs:
+        uplink = intf.strip()
+        pci_id = find_pci_id(uplink)
+        LOG.syslog("Uplink %s is PCI device %s", uplink, pci_id)
+
+        # check the card type, if x710, read lldp statusfrom nic
+        vendor = int(readfile("/sys/bus/pci/devices/%s/vendor" % pci_id), 16)
+        device = int(readfile("/sys/bus/pci/devices/%s/device" % pci_id), 16)
+
+        LOG.syslog("pci_id %s vendor %#04x device %#04x", pci_id, vendor, device)
+
+        # check if X710 NIC, if yes, read lldp status
+        if (vendor == 0x8086 and device == 0x1572):
+            # default assume status as lldp is stopped
+            lldp_status[uplink] = lldp_stop_str
+            if os.path.exists("/sys/bus/pci/devices/%s/driver" % pci_id):
+                driver = os.path.basename(os.readlink("/sys/bus/pci/devices/%s/driver" % pci_id))
+                if driver == "i40e":
+                    status = readfile("/sys/kernel/debug/%s/%s/command" %(driver, pci_id))
+                    # update the status
+                    if (status is lldp_start_str or status is lldp_stop_str):
+                        lldp_status[uplink] = status
+                    LOG.syslog("LLDP status read for Uplink %s pci_id %s vendor %#04x device %#04x", uplink, pci_id, vendor, device)
+            else:
+                LOG.syslog("LLDP status not read for Uplink %s pci_id %s vendor %#04x device %#04x as driver file doesn't exist", uplink, pci_id, vendor, device)
+
+
+# update the lldp tx status for interface
+def update_lldp_status_at_device(uplink, status)
```

Comment:
most of the code under `read_lldp_status_from_device` and `update_lldp_status_at_device` is the same, until you reach the part where you actually read/update.
maybe you can have a common method that returns a boolean that says `true` (and maybe tuple with more info if needed) when it is the desired device?